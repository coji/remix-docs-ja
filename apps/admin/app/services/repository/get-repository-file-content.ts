import { fromAsyncThrowable } from 'neverthrow'
import fs from 'node:fs/promises'
import path from 'node:path'
import { md5sum } from '~/libs/md5sum'
import type { RepositoryFile } from './types'

const fileContentImpl = async (
  directory: string,
  filename: string,
): Promise<RepositoryFile> => {
  const content = await fs.readFile(path.join(directory, filename), 'utf-8')
  const md5 = md5sum(content)

  return { filename, content, md5 }
}

export const getRepositoryFileContent = (
  directory: string,
  filename: string,
) => {
  return fromAsyncThrowable(
    () => fileContentImpl(directory, filename),
    (err) => err,
  )()
}
