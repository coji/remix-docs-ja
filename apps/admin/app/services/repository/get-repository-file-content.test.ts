import { describe, expect, test } from 'vitest'
import { getRepositoryFileContent } from './get-repository-file-content'

describe('getRepositoryFileContent', () => {
  test('should return the content of the file', async () => {
    const result = await getRepositoryFileContent(
      'test/fixtures/projects/projectA/docs',
      'index.md',
    )
    expect(result.isOk()).toBe(true)
    expect(result._unsafeUnwrap()).toMatchSnapshot()
  })

  test('should return an error if the file does not exist', async () => {
    const result = await getRepositoryFileContent(
      'test/fixtures/projects/projectA/docs',
      'non-existent.md',
    )
    expect(result.isErr()).toBe(true)
    expect(result._unsafeUnwrapErr()).toMatchSnapshot()
  })
})
