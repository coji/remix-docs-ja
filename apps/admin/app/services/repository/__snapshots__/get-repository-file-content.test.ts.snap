// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getRepositoryFileContent > should return an error if the file does not exist 1`] = `[Error: ENOENT: no such file or directory, open 'test/fixtures/projects/projectA/docs/non-existent.md']`;

exports[`getRepositoryFileContent > should return the content of the file 1`] = `
{
  "content": "---
title: Getting Started with Task Management Application
---

This document will guide you through the initial setup and basic usage of our Task Management Application. We will cover topics such as creating an account, logging in, creating and managing tasks, and collaborating with your team. By the end of this guide, you should have a good understanding of how to use the application to effectively manage your tasks and projects.
",
  "filename": "index.md",
  "md5": "b6ea662901a38626e3c5f45e532082ad",
}
`;
